### ISSUE TITLE
Search pattern not working correctly?

### ISSUE THREAD

**Issue by reijin90** [hugsy/gef#399]
Your issue will be closed unless you confirm the following:
  * [x] Did you use the latest version of GEF from `master` branch?
  * [x] Did you read the [documentation](https://gef.readthedocs.org/en/latest/) first?
  * [x] Did you check [closed issues](https://github.com/hugsy/gef/issues)?

### Step 1: Describe your environment

  * Operating System: Windows WSL -> Ubuntu VM
  * Architecture: x86_64
  * GDB version (including the Python library version): 

### Step 2: Describe your problem

search-pattern feels unreliable to me. it doesn't seem to like certain inputs and then doesn't find anything even if it is clearly there. Example:
```
gef➤  search-pattern \x60\xf0\x0f\x01
[+] Searching 'x60xf0x0fx01' in memory
gef➤  search-pattern '\x60\xf0\x0f\x01'
[+] Searching 'x60xf0x0fx01' in memory
gef➤  search-pattern "\x60\xf0\x0f\x01"
[+] Searching 'x60xf0x0fx01' in memory
gef➤  search-pattern 0x010ff060
[+] Searching '0x010ff060' in memory
[+] In '/home/[...]000), permission=r-x
  0x115ffa0 - 0x115ffb0  →   "\x60\xf0\x0f\x01[...]"
[+] In '/home/m[...]2000-0x1b8f000), permission=rw-
  0x1823098 - 0x18230a8  →   "\x60\xf0\x0f\x01[...]"
[+] In (0xc41f3d8000-0xc438500000), permission=rw-
  0xc420011d78 - 0xc420011d88  →   "\x60\xf0\x0f\x01[...]"
  0xc42001fd78 - 0xc42001fd88  →   "\x60\xf0\x0f\x01[...]"
  0xc42011e690 - 0xc42011e6a0  →   "\x60\xf0\x0f\x01[...]"
```

I'm expecting to find somthing with one of the char patterns, but it doesnt work - what is my or search-patterns mistake?

#### Expected results
search-pattern finding the given char pattern

**Comment by hugsy** [comment]
See https://gef.readthedocs.io/en/master/commands/search-pattern/

TL;DR : if you pass raw characters (i.e `\xff`) as argument, they **must** be protected (so `\xff` becomes `\\xff`)

```
gef➤  grep \x40\x07\xe2\xfe\xff\x7f                         # <---- WRONG WAY
[+] Searching 'x40x07xe2xfexffx7f' in memory

gef➤  grep \\x40\\x07\\xe2\\xfe\\xff\\x7f                   # <---- CORRECT WAY
[+] Searching '\x40\x07\xe2\xfe\xff\x7f' in memory
[+] In '/bin/ls'(0x61e000-0x61f000), permission=rw-
  0x61e1c0 - 0x61e1d8  →   "\x40\x07\xe2\xfe\xff\x7f[...]"

gef➤  xref 0x7ffffee20740                                   # also correct
[+] Searching '0x7ffffee20740' in memory
[+] In '/bin/ls'(0x61e000-0x61f000), permission=rw-
  0x61e1c0 - 0x61e1d8  →   "\x40\x07\xe2\xfe\xff\x7f[...]"
```
